### PR TITLE
run e2e tests with 2 guest worker nodes

### PR DESCRIPTION
**Pull request by nunnatsa**
**Release note**:

```release-note
None
```

### CODE REVIEW

File: kubevirtci [hunk 1]
```diff
@@ -7,7 +7,7 @@ export CAPK_GUEST_K8S_VERSION=${CAPK_GUEST_K8S_VERSION:-v1.26.0}
 export KUBEVIRTCI_TAG=${KUBEVIRTCI_TAG:-2309141019-029e67a}
 export KUBECONFIG=$(cluster-up/cluster-up/kubeconfig.sh)
 export KUBEVIRT_DEPLOY_PROMETHEUS=false
-export KUBEVIRT_NUM_NODES=${KUBEVIRT_NUM_NODES:-1}
+export KUBEVIRT_NUM_NODES=${KUBEVIRT_NUM_NODES:-3}
```
Comment (davidvossel) [inline kubevirtci:10]: why 3? that's a lot of resources. I don't know if our CI infra can do this

File: kubevirtci [hunk 2]
```diff
@@ -177,7 +177,7 @@ function kubevirtci::create_cluster() {
 
 	echo "Using cluster template $template"
 
-	$CLUSTERCTL_PATH generate cluster ${TENANT_CLUSTER_NAME} --target-namespace ${TENANT_CLUSTER_NAMESPACE} --kubernetes-version ${CAPK_GUEST_K8S_VERSION} --control-plane-machine-count=1 --worker-machine-count=1 --from $template | ${_kubectl} apply -f -
+	$CLUSTERCTL_PATH generate cluster ${TENANT_CLUSTER_NAME} --target-namespace ${TENANT_CLUSTER_NAMESPACE} --kubernetes-version ${CAPK_GUEST_K8S_VERSION} --control-plane-machine-count=1 --worker-machine-count=2 --from $template | ${_kubectl} apply -f -
```
Comment (davidvossel) [inline kubevirtci:180]: can we make the control plane node schedulable, so we don't need another worker?

**Comment by nunnatsa** [comment]
@davidvossel - fixed.

Could you PTAL?

File: hack/generate_clients.sh [hunk 3]
```diff
@@ -4,7 +4,6 @@ set -o nounset
 set -o pipefail
 
 go install k8s.io/code-generator/cmd/client-gen@latest
-go get kubevirt.io/api
```
Comment (davidvossel) [inline hack/generate_clients.sh:7]: why is this removed?

File: hack/run-k8s-e2e.sh [hunk 4]
```diff
@@ -135,9 +143,16 @@ if ./kubevirtci kubectl get storageprofile local; then
 fi
 }
 
+function make_control_plane_schedulable() {
```
Comment (davidvossel) [inline hack/run-k8s-e2e.sh:146]: it seems like this belongs in the kubevirtci script during the `create_cluster` function.

File: hack/cluster-sync-split.sh [hunk 5]
```diff
@@ -101,5 +101,8 @@ _kubectl apply --kustomize ./deploy/controller-infra/dev-overlay
 # ******************************************************
 # Wait for driver to rollout
 # ******************************************************
+_kubectl_tenant rollout restart ds/kubevirt-csi-node -n $CSI_DRIVER_NAMESPACE
+_kubectl rollout restart deployment/kubevirt-csi-controller -n $TENANT_CLUSTER_NAMESPACE
+
```
Comment (davidvossel) [inline hack/cluster-sync-split.sh:106]: why is this necessary?

File: hack/generate_clients.sh (hunk 3 above)
Comment (nunnatsa) [reply hack/generate_clients.sh:7]: This is not needed because kv api is already in go.mod

But more than that, after the release of kv v1.2.0, this line upgrades kv api package, and the git diff fails later.

File: hack/run-k8s-e2e.sh (hunk 4 above)
Comment (nunnatsa) [reply hack/run-k8s-e2e.sh:146]: This is not needed for the other e2e tests. I didn't want to change the default k8s setting where the changes are not on use.

File: hack/cluster-sync-split.sh (hunk 5 above)
Comment (nunnatsa) [reply hack/cluster-sync-split.sh:106]: The image name is not changed when it te-built. That why kubectl rollout status does nothing, and the pods are still running with the old imsges.

With this change, the pods will use the latest images that was just built a few lines above, letting you fix code and retry the test.

Comment (davidvossel) [reply hack/cluster-sync-split.sh:106]: oh nice! good addition
